### PR TITLE
Show floating prisoner with the active box

### LINKED ISSUE #5: Show floating prisoner when running iterations.

### PR DESCRIPTION
**Pull request by mohitkyadav**
Closes #5

### CODE REVIEW

**Comment by mohitkyadav** [comment]
apologies for the same commit message

File: src/components/Box/Box.tsx [hunk 1]
```diff
@@ -9,8 +9,10 @@ export const Box: React.FC<BoxProps> = ({ number, index }) => {
       id={`${index}-${number}`}
       className="main-page__play__boxes__grid__box"
     >
-      {number}
-      <div className="main-page__play__boxes__grid__box__cover">{index}</div>
+      {number + 1}
```
Comment (mohitkyadav) [inline src/components/Box/Box.tsx:12]: ditching index in favor of number

File: src/components/Box/Box.tsx [hunk 2]
```diff
@@ -9,8 +9,10 @@ export const Box: React.FC<BoxProps> = ({ number, index }) => {
...
+      <div className="main-page__play__boxes__grid__box__cover">
+        {index + 1}
```
Comment (mohitkyadav) [inline src/components/Box/Box.tsx:14]: same,

File: src/pages/MainPage/MainPage.scss [hunk 3]
```diff
@@ -32,6 +32,26 @@
         grid-gap: 4rem;
         margin-top: 1rem;
 
+        &__prisoner {
+          position: absolute;
+          width: 8rem;
+          height: 2.4rem;
+          border-radius: 0.5rem;
+          left: -8rem;
+          top: -0;
```
Comment (aashutoshrathi) [inline src/pages/MainPage/MainPage.scss:41]: ah?

File: src/pages/MainPage/MainPage.tsx [hunk 4]
```diff
@@ -33,6 +33,9 @@ export const MainPage: React.FC = () => {
     let boxIdx = manIdx;
     let boxCounter = 0;
     setGameRunning(true);
+    await new Promise((r) => setTimeout(r, cfg.intervalDur * 2));
```
Comment (aashutoshrathi) [inline src/pages/MainPage/MainPage.tsx:36]: Extract this out as a util function for delay or sleep like below and use it everywhere
```ts
    const sleep = async (delayInMs: number) => new Promise((r) => setTimeout(r, delayInMs));
```

File: src/pages/MainPage/MainPage.tsx [hunk 5]
```diff
@@ -128,6 +150,16 @@ export const MainPage: React.FC = () => {
             {boxes.map((box, index) => (
               <Box key={`${box}-${index}`} number={box} index={index} />
             ))}
+            <div
+              className={`main-page__play__boxes__grid__prisoner ${
+                gameRunning
+                  ? "main-page__play__boxes__grid__prisoner--active"
+                  : ""
+              }`}
+              id="floating-prisoner"
```
Comment (aashutoshrathi) [inline src/pages/MainPage/MainPage.tsx:159]: use `ref` over `id` in these cases, where you are using the same thing programmatically in code

File: src/pages/MainPage/MainPage.tsx [hunk 6]
```diff
@@ -87,6 +103,12 @@ export const MainPage: React.FC = () => {
 
       if (manIdx === newBoxes.length) {
         setGameRunning(false);
+
+        if (floatingPrisoner) {
+          floatingPrisoner.style.left = `${-100}px`;
```
Comment (aashutoshrathi) [inline src/pages/MainPage/MainPage.tsx:108]: Let's make a variable for this `100`

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]

File: src/pages/MainPage/MainPage.tsx (hunk 5 above)
Comment (mohitkyadav) [reply src/pages/MainPage/MainPage.tsx]: done

File: src/pages/MainPage/MainPage.tsx (hunk 6 above)
Comment (mohitkyadav) [reply src/pages/MainPage/MainPage.tsx]: Updated

File: src/pages/MainPage/MainPage.tsx [hunk 7]
```diff
@@ -128,6 +152,16 @@ export const MainPage: React.FC = () => {
...
             ))}
+            <div
+              className={`main-page__play__boxes__grid__prisoner ${
+                gameRunning
+                  ? "main-page__play__boxes__grid__prisoner--active"
+                  : ""
+              }`}
+              ref={floatingPrisonerRef}
+            >
+              {currentPrisoner + 1}
```
Comment (aashutoshrathi) [inline src/pages/MainPage/MainPage.tsx:163]: One Last optional feedback, maybe we should make it `P${currentPrisoner + 1}`, for distinction that this represents Prisoner

**Review by aashutoshrathi — APPROVED** [review]

File: src/pages/MainPage/MainPage.tsx (hunk 7 above)
Comment (aashutoshrathi) [reply src/pages/MainPage/MainPage.tsx:163]: Similarly if you want we can add `B` as prefix on Boxes

Comment (mohitkyadav) [reply src/pages/MainPage/MainPage.tsx:163]: Arigato! Will add this in next PR

**Pull request merged**